src/components/data-exploration/generic-gene-table/GeneTable.jsx

```diff
@@ -7,47 +7,71 @@ import _ from 'lodash';
 import PropTypes from 'prop-types';
 import { changeGeneSelection } from 'redux/actions/genes';
 import GeneSelectionStatus from 'redux/actions/genes/geneSelectionStatus';
-import { geneTableUpdateReason } from 'utils/geneTable/geneTableUpdateReason';
 import FocusButton from 'components/FocusButton';
 import PlatformError from 'components/PlatformError';
-import useLazyEffect from 'utils/customHooks/useLazyEffect';
 import GeneSelectionMenu from 'components/data-exploration/generic-gene-table/GeneSelectionMenu';
 import FilterGenes from 'components/data-exploration/generic-gene-table/FilterGenes';
 import Loader from 'components/Loader';
+import { sanitizeString } from 'components/data-exploration/generic-gene-table/FilterGenes';
+
+const valueComparator = (key) => (a, b) => {
+  if (typeof a[key] === 'string') return a[key].localeCompare(b[key]);
+  if (typeof a[key] === "number") return a[key] - b[key];
+  return 0;
+}
 
 const GeneTable = (props) => {
   const {
-    experimentId, onUpdate, error, loading, columns, data,
+    experimentId, error, loading, columns, propData, loadData,
     total, initialTableState, width, height, extraOptions,
   } = props;
 
   const dispatch = useDispatch();
   const selectedGenes = useSelector((state) => state.genes.selected);
   const [geneNameFilterState, setGeneNameFilterState] = useState({});
 
+  const tableStateAllEntries = {
+    pagination: {
+      current: 1,
+      pageSize: 1000000,
```

Comment:
What is this hardcoded page size 1 million? do we even need it? I think you can use total here to know the page size?